main/java/nearsoft/academy/bigdata/recommendation/MovieRecommender.java

```diff
@@ -0,0 +1,220 @@
+                    //between the parenthesis. In this case, is the product ID:
+                    String ProductID = MatchProduct.group(1);
+
+                    //Check if the current product is already in the hashmap, if it's not
+                    //sum one to the number of products
+                    if(!Products.containsKey(ProductID)) {
+
+                        this.TotalProducts += 1;
+                        //Now we save the key and product in the hashmap
+                        Products.put(ProductID, this.TotalProducts);
+                    }
+
+                    //Use this variable to later use on the CSV file
+                    SProduct = Products.get(ProductID).toString();
+
+                    //Need the user line of the file, so we scroll in the file until the reader finds it
+                    while(!MatchUser.matches()) {
+                        strCurrentLine = br.readLine();
+                        MatchUser = PatternUser.matcher(strCurrentLine);
+                    }
+
+                    //Now that the line has the user, use the same technique used in product
+                    String UserID = MatchUser.group(1);
+                    if(!Users.containsKey(UserID)){
+
+                        this.TotalUsers += 1;
+                        Users.put(UserID, this.TotalUsers);
+
+                    }
+                    //Use this variable to later use on the CSV file
+                    SUser = Users.get(UserID).toString();
+
+                    //Finally, do the same to the reviews
+                    while(!MatchReview.matches()){
+                        strCurrentLine = br.readLine();
+                        MatchReview = PatternReview.matcher(strCurrentLine);
+
+                    }
+
+                    String Review = MatchReview.group(1);
+
+                    //Now that the info is in the variables, we write it in the CSV file
+                    bufferedWriter.write(SUser + "," + SProduct + "," + Review + "\n");
+                }
+            }
+
+            bufferedWriter.close();
+            fileWriter.close();
+            inputStream.close();
+            br.close();
+
+        }
+
+        catch (FileNotFoundException e){
+
+            System.out.println("The file introduced cannot be found.");
+
+        }
+
+    }
+
+    //Now we give the results:
+    public int getTotalReviews(){
+        return this.TotalReviews;
+    }
+
+    public int getTotalProducts(){
+        return this.TotalProducts;
+    }
+
+    public int getTotalUsers(){
+        return this.TotalUsers;
+    }
+
+    //Since the hashmap for products takes the ID as the key and
+    //not the value, in recommendations we call this class to
+    //obtain the real value of the ID and return it to add it
+    //to the list
+    private String getProductID(int ID) {
+        for (String key : Products.keySet()) {
+            if (Products.get(key)==ID) {
```

Comment:
When comparing strings is better to use `.equals()` method instead of the == comparator